### ISSUE TITLE
Add extend to the type definition file

### ISSUE THREAD

**Issue by alex3165** [Khan/aphrodite#172]
The latest type definition file for Typescript doesn't expose `StyleSheet.extend` which is problematic because then we lose all the types when using extensions.

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/aphrodite/aphrodite.d.ts#L18

**Comment by xymostech** [comment]
Hello @alex3165! We didn't create that file, so I'm not sure how they get updated. Are they just manually created and maintained? I could just make a PR to them if that's the case.

**Comment by xymostech** [comment]
I sent in a PR: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/12845

**Comment by alex3165** [comment]
Hi @xymostech, I have check your type definition file and tested it in my project and it works perfectly. Line 94 you redefined the `StyleSheetStatic ` interface with :
```
    interface StyleSheetStatic {
        extend(extensions: Extension[]): Exports;
    }
```

I thought it would override the previously defined `StyleSheetStatic` :
```
    interface StyleSheetStatic {
        /**
         * Create style sheet
         */
        create<T extends StyleDeclaration>(styles: T): T;
        /**
         * Rehydrate class names from server renderer
         */
        rehydrate(renderedClassNames: string[]): void;
    }
```

But it look like it is working so I guess it does a merge of the 2 interface.

**Comment by xymostech** [comment]
Glad that the types worked for you!

Yeah, the [docs about interface declarations](https://www.typescriptlang.org/docs/handbook/declaration-merging.html#merging-interfaces) say that if you have two declarations, they will merge together. I did this because `Exports` refers to `StyleSheetStatic` but also needs to be used in `StyleSheetStatic`, so I thought this made that make sense.

**Comment by upta** [comment]
Looks like the PR for this ended up in purgatory?  Also, would it be possible to add an overload to StyleSheetStatic->create like:
```
create( styles: StyleDeclaration ): StyleDeclaration;
```

in addition to the existing
```
create<T extends StyleDeclaration>( styles: T ): T;
```

The later plays havoc with intellisense unless you're explicitly importing/using StyleDeclaration.

If I can help in any way let me know, thanks!

**Comment by lencioni** [comment]
I'm not a TypeScript expert, but it seems that this was fixed by #302. Please open a new issue or PR if I am wrong!
